### DISCUSSION TITLE
Angular component testing can't find stylesheet to import

### DISCUSSION THREAD

**Discussion by Avcajaraville** [cypress-io/cypress#29962] [Questions and Help] [answered]
I am using angular 18 and cypress 13.

When running my component testing specs, I cant find a way of properly loading the sass files I need.

For context, I am using angular material, and my tree setup looks something like this:

```
/src
  |-> /app
  |-> /sass
    |-> /sass
            |-> /base
            |-> /mixins
            |-> ...
            |-> styles.scss
/cypress
cypress.config.ts
```

My cypress config is:

```typescript
export default defineConfig({
  component: {
    devServer: {
      framework: 'angular',
      bundler: 'webpack',
      options: {
        projectConfig: {
          root: '',
          sourceRoot: '',
          buildOptions: {
            styles: ['./src/sass/styles.scss'],
          },
        },
      },
    },
    specPattern: '**/*.cy.ts',
  },
});
```

Component testing via Cypress works great, however, I am having problems loading the sass file that I need.

Basically, sass compiler is having issue when looking for the right path when using a `@use` rule.

I get this error:

```
Can't find stylesheet to import.
  ╷
1 │ @use 'mixins';
  │ ^^^^^^^^^^^^^
```

Which, by the way, doesn't happens when using angular, it only happens when using cypress component testing.

Any idea what might be the issue here?

Thanks!

**Comment by atomicrobokid** [comment]
Think i'm facing the same issue as you.

I have an NX project and one of the libs is purely styles. Other libs import those styles. Libs project.json is configured with:
```
      "stylePreprocessorOptions": {
        "includePaths": [
          "libs/styles/src/lib/sass/**"
        ]
      },
```

This means that I can shorten the import urls as below:
This:
`@use 'main/index' as *;`
Instead of 
`@use 'libs/styles/src/lib/sass/main/index' as *;`

Which helps when the libs are bundled and published.

Everything works great, except Cypress, no matter what options i try to configure in `project.json` or `cypress.config.ts`

Disabling the cypress component tests for now

**Comment by rene-anderes** [comment]
I have the same problem.
Angular 18, Cypress 13.17

**cypress.config.ts:**

```
export default defineConfig({
  component: {
    devServer: {
      framework: "angular",
      bundler: "webpack",
      options: {
        projectConfig: {
          root: "",
          sourceRoot: "src",
          buildOptions: {
            outputPath: "dist/browser",
          },
        },
      },
    },
    specPattern: "**/*.cy.ts",
  },
});
```


**angular.json**

```
"stylePreprocessorOptions": {
                            "includePaths": [
                                "src/styles",
                                "."
                            ]
                        }

```

**Fehler**

```
Module build failed (from ./node_modules/sass-loader/dist/cjs.js):
Can't find stylesheet to import.

  ╷
1 │ @use 'variables' as *;
  │ ^^^^^^^^^^^^^^^^^^^^^
  ╵

```

The Angular application works, only the Cypress component tests do not work. Is this a Cypress bug or a problem with the configuration?

**Comment by mariavannman** [comment]
I have the same problem in our nx mono repo.
Angular 19 Cypress 14.3.8.
Everything works fine except when testing e2e-watch.

An uncaught error was detected outside of a test failed
No commands were issued in this test.
Error
The following error originated from your test code, not from Cypress.

> Module build failed (from ./node_modules/sass-loader/dist/cjs.js):
Can't find stylesheet to import.
╷
1 │ @use 'abstracts' as *;
│ ^^^^^^^^^^^^^^^^^^^^^

**Comment by jennifer-shehane** [comment, marked as answer]
When you pass a custom options.projectConfig in cypress.config, it replaces the configuration Cypress auto-detects from angular.json — so the stylePreprocessorOptions.includePaths your Sass relies on no longer apply, and @use 'mixins' can’t be resolved.

Add the same include paths to your buildOptions:

```js
buildOptions: {
  styles: ['src/styles.scss'],
  stylePreprocessorOptions: {
    includePaths: ['src/styles'],
  },
},
```

the docs will be updated for this.
